sdk/dashboard/Azure.ResourceManager.Dashboard/src/Generated/GrafanaResourceCollection.cs

```diff
@@ -0,0 +1,387 @@
+    {
+        private readonly ClientDiagnostics _grafanaResourceGrafanaClientDiagnostics;
+        private readonly GrafanaRestOperations _grafanaResourceGrafanaRestClient;
+
+        /// <summary> Initializes a new instance of the <see cref="GrafanaResourceCollection"/> class for mocking. </summary>
+        protected GrafanaResourceCollection()
+        {
+        }
+
+        /// <summary> Initializes a new instance of the <see cref="GrafanaResourceCollection"/> class. </summary>
+        /// <param name="client"> The client parameters to use in these operations. </param>
+        /// <param name="id"> The identifier of the parent resource that is the target of operations. </param>
+        internal GrafanaResourceCollection(ArmClient client, ResourceIdentifier id) : base(client, id)
+        {
+            _grafanaResourceGrafanaClientDiagnostics = new ClientDiagnostics("Azure.ResourceManager.Dashboard", GrafanaResource.ResourceType.Namespace, DiagnosticOptions);
+            TryGetApiVersion(GrafanaResource.ResourceType, out string grafanaResourceGrafanaApiVersion);
+            _grafanaResourceGrafanaRestClient = new GrafanaRestOperations(_grafanaResourceGrafanaClientDiagnostics, Pipeline, DiagnosticOptions.ApplicationId, BaseUri, grafanaResourceGrafanaApiVersion);
+#if DEBUG
+			ValidateResourceId(Id);
+#endif
+        }
+
+        internal static void ValidateResourceId(ResourceIdentifier id)
+        {
+            if (id.ResourceType != ResourceGroup.ResourceType)
+                throw new ArgumentException(string.Format(CultureInfo.CurrentCulture, "Invalid resource type {0} expected {1}", id.ResourceType, ResourceGroup.ResourceType), nameof(id));
+        }
+
+        /// <summary>
+        /// Create or update a workspace for Grafana resource. This API is idempotent, so user can either create a new grafana or update an existing grafana.
+        /// Request Path: /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Dashboard/grafana/{workspaceName}
+        /// Operation Id: Grafana_Create
+        /// </summary>
+        /// <param name="waitForCompletion"> Waits for the completion of the long running operations. </param>
+        /// <param name="workspaceName"> The name of Azure Managed Grafana. </param>
+        /// <param name="body"> The GrafanaResource to use. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <exception cref="ArgumentException"> <paramref name="workspaceName"/> is an empty string, and was expected to be non-empty. </exception>
+        /// <exception cref="ArgumentNullException"> <paramref name="workspaceName"/> is null. </exception>
+        public async virtual Task<ArmOperation<GrafanaResource>> CreateOrUpdateAsync(bool waitForCompletion, string workspaceName, GrafanaResourceData body = null, CancellationToken cancellationToken = default)
```

Comment:
they forgot to add `required: true` for body parameter

Reply:
I will ask service to make the change from swagger